server/src/main/java/org/elasticsearch/index/mapper/MapperFeatures.java

```diff
@@ -35,7 +35,8 @@ public Set<NodeFeature> getFeatures() {
             ObjectMapper.SUBOBJECTS_AUTO,
             KeywordFieldMapper.KEYWORD_NORMALIZER_SYNTHETIC_SOURCE,
             SourceFieldMapper.SYNTHETIC_SOURCE_STORED_FIELDS_ADVANCE_FIX,
-            Mapper.SYNTHETIC_SOURCE_KEEP_FEATURE
+            Mapper.SYNTHETIC_SOURCE_KEEP_FEATURE,
+            SourceFieldMapper.SYNTHETIC_SOURCE_RESTRICTIONS_REMOVED
```

Comment:
`SYNTHETIC_SOURCE_WITH_COPY_TO_AND_DOC_VALUES_FALSE_SUPPORT`? I Know it is longer but much more descriptive.